pages/products/index.tsx

```diff
@@ -29,6 +29,23 @@ type ProductListingProps = Readonly<{
   products: ReadonlyArray<Product> | undefined,
 }>
 
+const createQueryStringUrl = (params: Record<string, string | string[] | undefined>) =>
+  Object.keys(params).length > 0 ?
+    Object.entries(params).reduce(
```

Comment:
You can use the same approach, as you use for the array values, for all the params. Just map a param-creating function over all the entries, join them with an `"&"` and prepend a `"?"` if the result is not empty. I believe it would be a bit simpler than the reduce.